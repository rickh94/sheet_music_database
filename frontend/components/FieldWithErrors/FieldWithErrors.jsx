import React from 'react'
import PropTypes from 'prop-types'

import { Field, Control, Label, Help } from 'react-bulma-components/lib/components/form'

export default function FieldWithErrors(props) {
  return (
    <Field className={props.className}>
      {props.label && <Label>{props.label}</Label>}
      <Control className="is-expanded">{props.children}</Control>
      {props.error && <Help color="danger">{props.error}</Help>}
    </Field>
  )
}

FieldWithErrors.propTypes = {
  className: PropTypes.string,
  label: PropTypes.string,
  children: PropTypes.any,
  error: PropTypes.any
}
