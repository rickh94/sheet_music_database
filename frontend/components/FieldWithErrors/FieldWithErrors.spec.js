import React from 'react'
import { shallow, mount } from 'enzyme'
import FieldWithErrors from './FieldWithErrors'

import '../../setupTests'

describe('<FieldWithErrors />', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(<FieldWithErrors />)
  })

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('renders a Field and a Control', () => {
    expect(wrapper.exists('Field')).toBeTruthy()
    expect(wrapper.exists('Control')).toBeTruthy()
  })

  it('renders label from props', () => {
    const label = 'test label'
    wrapper.setProps({ label })
    expect(wrapper.exists('Label')).toBeTruthy()
    expect(wrapper.find('Label').contains(label)).toBeTruthy()
  })

  it('renders children', () => {
    const otherWrapper = shallow(
      <FieldWithErrors>
        <div>test</div>
      </FieldWithErrors>
    )
    expect(otherWrapper.exists('div')).toBeTruthy()
    expect(otherWrapper.find('div').contains('test')).toBeTruthy()
  })

  it('renders errors', () => {
    const error = 'test error'
    wrapper.setProps({error})
    expect(wrapper.exists('Help')).toBeTruthy()
    expect(wrapper.find('Help').contains(error)).toBeTruthy()
  })

  it('does not render label if not provided', () => {
    wrapper.setProps({label: null})
    expect(wrapper.exists('Label')).toBeFalsy()
  })
})
