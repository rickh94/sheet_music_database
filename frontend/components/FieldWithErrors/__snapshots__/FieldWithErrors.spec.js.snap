// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FieldWithErrors /> matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FieldWithErrors />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "align": null,
      "backgroundColor": undefined,
      "children": Array [
        undefined,
        <Control
          className="is-expanded"
          fullwidth={false}
          iconLeft={false}
          iconRight={false}
          loading={false}
          renderAs="div"
          size={null}
          style={Object {}}
        />,
        undefined,
      ],
      "className": "",
      "clearfix": undefined,
      "clipped": undefined,
      "hidden": undefined,
      "horizontal": false,
      "invisible": undefined,
      "italic": undefined,
      "kind": null,
      "marginless": undefined,
      "multiline": false,
      "overlay": undefined,
      "paddingless": undefined,
      "pull": undefined,
      "radiusless": undefined,
      "renderAs": "div",
      "responsive": undefined,
      "shadowless": undefined,
      "style": Object {},
      "textAlignment": undefined,
      "textColor": undefined,
      "textSize": undefined,
      "textTransform": undefined,
      "textWeight": undefined,
      "unselectable": undefined,
    },
    "ref": null,
    "rendered": Array [
      undefined,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "backgroundColor": undefined,
          "children": null,
          "className": "is-expanded",
          "clearfix": undefined,
          "clipped": undefined,
          "fullwidth": false,
          "hidden": undefined,
          "iconLeft": false,
          "iconRight": false,
          "invisible": undefined,
          "italic": undefined,
          "loading": false,
          "marginless": undefined,
          "overlay": undefined,
          "paddingless": undefined,
          "pull": undefined,
          "radiusless": undefined,
          "renderAs": "div",
          "responsive": undefined,
          "shadowless": undefined,
          "size": null,
          "style": Object {},
          "textAlignment": undefined,
          "textColor": undefined,
          "textSize": undefined,
          "textTransform": undefined,
          "textWeight": undefined,
          "unselectable": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      undefined,
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "align": null,
        "backgroundColor": undefined,
        "children": Array [
          undefined,
          <Control
            className="is-expanded"
            fullwidth={false}
            iconLeft={false}
            iconRight={false}
            loading={false}
            renderAs="div"
            size={null}
            style={Object {}}
          />,
          undefined,
        ],
        "className": "",
        "clearfix": undefined,
        "clipped": undefined,
        "hidden": undefined,
        "horizontal": false,
        "invisible": undefined,
        "italic": undefined,
        "kind": null,
        "marginless": undefined,
        "multiline": false,
        "overlay": undefined,
        "paddingless": undefined,
        "pull": undefined,
        "radiusless": undefined,
        "renderAs": "div",
        "responsive": undefined,
        "shadowless": undefined,
        "style": Object {},
        "textAlignment": undefined,
        "textColor": undefined,
        "textSize": undefined,
        "textTransform": undefined,
        "textWeight": undefined,
        "unselectable": undefined,
      },
      "ref": null,
      "rendered": Array [
        undefined,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "backgroundColor": undefined,
            "children": null,
            "className": "is-expanded",
            "clearfix": undefined,
            "clipped": undefined,
            "fullwidth": false,
            "hidden": undefined,
            "iconLeft": false,
            "iconRight": false,
            "invisible": undefined,
            "italic": undefined,
            "loading": false,
            "marginless": undefined,
            "overlay": undefined,
            "paddingless": undefined,
            "pull": undefined,
            "radiusless": undefined,
            "renderAs": "div",
            "responsive": undefined,
            "shadowless": undefined,
            "size": null,
            "style": Object {},
            "textAlignment": undefined,
            "textColor": undefined,
            "textSize": undefined,
            "textTransform": undefined,
            "textWeight": undefined,
            "unselectable": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        undefined,
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
